### ISSUE TITLE
attribute dependency cycle

### ISSUE THREAD

**Issue by hugeshi** [AbsaOSS/spline-spark-agent#125]
## Describe the bug
can't generate the lineage info by accessing attribute-lineage-and-impact

## Versions
Spline 0.6.0-SNAPSHOT
Spark 2.3.2
Scala 2.11.6

## Components State
- [ yes] ArangoDB running without errors
- [ yes] ArangoDB spline database initialized
- Rest Gateway running and
  - [ yes] connects to ArangoDB
  - [ yes] there are no errors in logs
- Spline UI running and
  - [ yes] connects to Rest Gateway consumer
  - [yes ] there are no errors in logs

## To Reproduce
Steps to reproduce the behavior OR commands run:
1. Go to 'lineage-detailed'
2. Click on ' InsertIntoHiveTable' node
3. click value 'attribute name'

## Expected behaviour
the clicked attribute lineage info can be display in the graph

## Screenshots
![image](https://user-images.githubusercontent.com/6722878/83851989-b6d1b480-a745-11ea-9a50-81ac8d24711b.png)
![image](https://user-images.githubusercontent.com/6722878/83852034-c3eea380-a745-11ea-9b84-c03704343209.png)

## Desktop (please complete the following information):
- OS: [iOS]
- Browser [chrome]
- Version [22]

**Comment by hugeshi** [comment]
what's the priority of this issue, what's the timeline for 0.5.5 version?

**Comment by wajda** [comment]
We should be able to get to it on Monday. We still have yet to understand the impact of this issue on the rest of the system. `0.5.5` is planned to be released by the end of the next week. If this fix can't make it to `0.5.5` it'll go to the next possible release.

**Comment by cerveada** [comment]
I found the cause of the issue. Spark's `AttributeReference` has attribute `qualifier: Seq[String]`. This attribute is empty, but not always and for example when I create tempView to access dataframe from spark sql the qualifier will contain the name of the view. It's still the same attribute with same id, but when it's taken from the view the qualifier is different. As a consequence the hashcode and equals are different too.

Since we care about where the data are taken from and not how the user requested them (using view or not) we should ignore the qualifier. There is already ExprId in the attribute that I was planing to use in Spline 0.6 So I think we should change the cache to use the ExprId.

The issue is reproducible using this code:
```
  testData.createOrReplaceTempView("testDataTable")
  val result1 = spark.sql("SELECT a.*,ROW_NUMBER() OVER (PARTITION BY a.category ORDER BY a.revenue ASC) as row from testDataTable a")
  result1.createOrReplaceTempView("result1Table")
  val result2 = spark.sql("select a.* from result1Table a where row=1")
```

**Comment by cerveada** [comment]
Hello @hugeshi,
Fix to the issue is already in [release/0.5 branch](https://github.com/AbsaOSS/spline-spark-agent/tree/release/0.5). Could you test the agent form there and let us know if it solves the issue?

**Comment by wajda** [comment]
I'm counting it to be fixed and releasing 0.5.5.
